### ISSUE TITLE
HttpClient: HttpRedirect feature worked incorectly

### ISSUE THREAD

**Issue by FTOH** [ktorio/ktor#610]
```kotlin
val httpClient = HttpClient(Apache) {
    install(HttpRedirect)
    install(HttpCookies) {
        default {
            runBlocking { addCookie("https://httpbin.org/", Cookie("test", "123")) }
        }
    }
    engine {
        customizeClient {
            setProxy(HttpHost("localhost", 8888))

            setSSLContext(SSLContexts.custom().loadTrustMaterial(null) { _, _ -> true }.build())
            setSSLHostnameVerifier(NoopHostnameVerifier())
        }
    }
}
println(httpClient.post<String>("https://httpbin.org/redirect-to") {
    url.parameters.append("url", "https://httpbin.org/get?other=param")
})
```
[![image](https://user-images.githubusercontent.com/5018813/46123609-ab787700-c249-11e8-980a-096289902742.png)](https://user-images.githubusercontent.com/5018813/46123609-ab787700-c249-11e8-980a-096289902742.png)

**Comment by e5l** [comment]
Hi @FTOH, thanks for the report.
I'll fix it in the next minor release.

**Comment by e5l** [comment]
Fixed in `1.0.0-beta-1`
